summarization/main.py

```diff
@@ -89,75 +90,51 @@ def main():
-        llm_model = os.getenv("LLM_MODEL", "mistral")
-        top_k = int(os.getenv("TOP_K", "12"))
-        citation_count = int(os.getenv("CITATION_COUNT", "12"))
-        retry_max_attempts = int(os.getenv("RETRY_MAX_ATTEMPTS", "3"))
-        retry_backoff_seconds = int(os.getenv("RETRY_BACKOFF_SECONDS", "5"))
+        # Load configuration from schema with typed access
+        config = load_typed_config("summarization")
         
         # Create adapters
         logger.info("Creating message bus publisher...")
         publisher = create_publisher(
-            message_bus_type=message_bus_type,
-            host=message_bus_host,
-            port=message_bus_port,
-            username=message_bus_user,
-            password=message_bus_password,
+            message_bus_type=config.message_bus_type,
+            host=config.message_bus_host,
+            port=config.message_bus_port,
+            username=config.message_bus_user,
+            password=config.message_bus_password,
         )
         
         logger.info("Creating message bus subscriber...")
         subscriber = create_subscriber(
-            message_bus_type=message_bus_type,
-            host=message_bus_host,
-            port=message_bus_port,
-            username=message_bus_user,
-            password=message_bus_password,
+            message_bus_type=config.message_bus_type,
+            host=config.message_bus_host,
+            port=config.message_bus_port,
+            username=config.message_bus_user,
+            password=config.message_bus_password,
         )
         
         logger.info("Creating document store...")
         document_store = create_document_store(
-            store_type=doc_store_type,
-            host=doc_store_host,
-            port=doc_store_port,
-            database=doc_store_name,
-            username=doc_store_user if doc_store_user else None,
-            password=doc_store_password if doc_store_password else None,
+            store_type=config.doc_store_type,
+            host=config.doc_store_host,
+            port=config.doc_store_port,
+            database=config.doc_store_name,
+            username=config.doc_store_user if config.doc_store_user else None,
+            password=config.doc_store_password if config.doc_store_password else None,
         )
         
         logger.info("Creating vector store...")
         vector_store = create_vector_store(
-            store_type=vector_store_type,
-            host=vector_store_host if vector_store_type != "inmemory" else None,
-            port=vector_store_port if vector_store_type != "inmemory" else None,
-            collection_name=vector_store_collection,
+            store_type=config.vector_store_type,
+            host=config.vector_store_host if config.vector_store_type != "inmemory" else None,
+            port=config.vector_store_port if config.vector_store_type != "inmemory" else None,
+            collection_name=config.vector_store_collection,
```

Comment:
Fixed: removed hardcoded `"message_embeddings"` and now use `config.vector_store_collection`. Added `vector_store_collection` to `documents/schemas/configs/summarization.json` with env var `VECTOR_DB_COLLECTION` and default `message_embeddings`.